components/studio/libexec/hab-studio-type-bare.sh

```diff
@@ -71,24 +79,25 @@ finish_setup() {
   studio_path="$full_path"
   studio_enter_command="${busybox_path}/bin/sh --login"
 
-  $bb mkdir -p $v $HAB_STUDIO_ROOT$HAB_ROOT_PATH/bin
+  # shellcheck disable=2086,2154
```

Comment:
```
In ./components/studio/libexec/hab-studio-type-bare.sh line 83:
  $bb mkdir -p $v "$HAB_STUDIO_ROOT""$HAB_ROOT_PATH"/bin
               ^-- SC2086: Double quote to prevent globbing and word splitting.
```
This one is a bit subtle.

Normally the right approach to `$v` which may be unset, empty or `-v` would be to put it in an array so it can be correctly expanded to something or nothing like so:
```bash
mkdir "${options[@]}" /tmp/foo
```
The problem with quoting the scalar is
```
$ v=
$ mkdir "$v" /tmp/foo
mkdir: cannot create directory ‘’: No such file or directory
```

But, as I understand it, all the `hab-studio-type-*.sh` files are [sourced here from hab-studio.sh](https://github.com/habitat-sh/habitat/blob/master/components/studio/bin/hab-studio.sh#L520), which means that the code inside the `hab-studio-type-*.sh` files is also necessarily POSIX.

If that's correct, I think we should add the appropriate `#!/bin/sh` and fix up the non-POSIX code in these files. We'd have to leave the `$v` unquoted though since I haven't found a clean way to deal with that apart from arrays, which aren't supported in POSIX `sh`.